### PR TITLE
[spinel] remove fetching the ncp dataset

### PR DESCRIPTION
**Pull request by zhanglongxia**
Fetching the dataset from NCP is added by the commit https://github.com/openthread/openthread/pull/4446. It is used to migrate the OpenThread from the NCP architecture to RCP architecture. This feature is not needed any more, this commit removes it from OpenThread.

### CODE REVIEW

File: src/lib/spinel/radio_spinel_impl.hpp [hunk 1]
```diff
@@ -45,13 +44,10 @@
 #include "common/encoding.hpp"
 #include "common/instance.hpp"
 #include "common/new.hpp"
-#include "common/settings.hpp"
 #include "lib/platform/exit_code.h"
 #include "lib/spinel/radio_spinel.hpp"
 #include "lib/spinel/spinel.h"
 #include "lib/spinel/spinel_decoder.hpp"
-#include "meshcop/dataset.hpp"
-#include "meshcop/meshcop_tlvs.hpp"
```
Comment (abtink) [inline src/lib/spinel/radio_spinel_impl.hpp:54]: I am happy that we remove this. Specially that we no longer have dependency in `radio_spinel` module to these core internal headers `meshcop_tlvs.hpp` and `dataset.hpp` :)

File: src/posix/platform/radio.cpp [hunk 2]
```diff
@@ -88,7 +88,6 @@ Radio::Radio(const char *aUrl)
 void Radio::Init(void)
 {
     bool        resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
-    bool        restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
```
Comment (abtink) [inline src/posix/platform/radio.cpp:91]: Maybe we can add a check that if URL happens to contain the `"ncp-dataset"` we print an error (and/or exit) indicating that this feature is removed?

**Review by abtink — APPROVED** [review]
Thanks @zhanglongxia for helping with this. LGTM. 👍 
I am happy to see that we remove this.

File: src/posix/platform/radio.cpp (hunk 2 above)
Comment (zhanglongxia) [reply src/posix/platform/radio.cpp]: Updated

**Review by bukepo — APPROVED** [review]

**Comment by jwhui** [comment]
Need to remove this line as well:
https://github.com/openthread/openthread/blob/9cb667c01fde61459afedab7c60aeaa28d60970a/.github/workflows/posix.yml#L321

**Comment by zhanglongxia** [comment]
> Need to remove this line as well:
> 
> https://github.com/openthread/openthread/blob/9cb667c01fde61459afedab7c60aeaa28d60970a/.github/workflows/posix.yml#L321

Updated

**Pull request merged**